File: familyChain-backend/internal/models/reward.go
package models

import "time"

// Reward 表示家长创建的实物奖励
type Reward struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	FamilyID    uint      `json:"family_id" gorm:"not null;index"`
	Name        string    `json:"name" gorm:"not null;size:255"`
	Description string    `json:"description" gorm:"type:text"`
	ImageURL    string    `json:"image_url" gorm:"type:text"`
	TokenPrice  int       `json:"token_price" gorm:"not null"`
	CreatedBy   uint      `json:"created_by" gorm:"not null"`
	Active      bool      `json:"active" gorm:"default:true;index"`
	Stock       int       `json:"stock" gorm:"default:1"`
	CreatedAt   time.Time `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt   time.Time `json:"updated_at" gorm:"autoUpdateTime"`
}

// RewardCreateRequest 表示创建奖品的请求
type RewardCreateRequest struct {
	Name        string `json:"name" binding:"required"`
	Description string `json:"description"`
	ImageURL    string `json:"image_url"`
	TokenPrice  int    `json:"token_price" binding:"required,min=1"`
	Stock       int    `json:"stock" binding:"min=0"`
}

// RewardUpdateRequest 表示更新奖品的请求
type RewardUpdateRequest struct {
	Name        string `json:"name"`
	Description string `json:"description"`
	ImageURL    string `json:"image_url"`
	TokenPrice  int    `json:"token_price" binding:"min=1"`
	Active      *bool  `json:"active"`
	Stock       int    `json:"stock" binding:"min=0"`
}

File: familyChain/src/types/reward.ts
export interface Reward {
  id: number;
  family_id: number;
  name: string;
  description: string;
  image_url: string;
  token_price: number;
  created_by: number;
  active: boolean;
  stock: number;
  created_at: string;
  updated_at: string;
  blockchain_id?: number;
}

export interface RewardCreateRequest {
  name: string;
  description: string;
  image_url: string;
  token_price: number;
  stock: number;
  blockchain_id?: number;
  create_on_blockchain?: boolean;
}

export interface RewardUpdateRequest {
  name?: string;
  description?: string;
  image_url?: string;
  token_price?: number;
  active?: boolean;
  stock?: number;
  blockchain_id?: number;
}

export type ExchangeStatus = 'pending' | 'completed' | 'cancelled' | 'confirmed' | 'failed';

export interface Exchange {
  id: number;
  reward_id: number;
  child_id: number;
  token_amount: number;
  status: ExchangeStatus;
  exchange_date: string;
  completed_date?: string;
  notes?: string;
  created_at: string;
  updated_at: string;
  reward_name?: string;
  reward_image?: string;
  child_name?: string;
}

export interface ExchangeUpdateRequest {
  status: ExchangeStatus;
  notes?: string;
} 